chaos-days/blog/2024-10-14-Optimizing-cluster-sizing-using-a-real-world-benchmark/index.md

```diff
@@ -0,0 +1,190 @@
+
+### Actual
+
+#### Minimal Requirements for our Cluster
+
+Based on known customer usage, and our own previous experiments, we 
+determined that the new cluster would need to create and complete a 
+baseline of 100 tasks per second, or about 8.6 million tasks per day.
+
+Other metrics that we want to preserve and keep track are the backpressure 
+to preserve user experience, guarantee that exporting speed can keep up 
+with the processing speed, write-to-import latency which tells us how long 
+it takes for a record to be written to being imported by our other 
+apps such as the operator.
+
+#### Reverse Engineering the Cluster Configuration
+
+For our new configurations the only resources that we are going to change 
+are the ones relevant to the factors described above. These are the 
+resources allocated to our zeebe-brokers, gateway and elasticSearch.
+
+Our starting point in resources was the configuration for our G3-S HA Plan
+as this already had the capability to significantly outperform the current 
+goal of 100 tasks per second. 
+
+The next step was to deploy our realistic benchmark, with a payload of 5 
+customer disputes per instance and start 7 instances per second, this 
+generated approximately 120 tasks per second (some buffer was added to guarantee performance).
+
+After this we reduced the resources iteratively until we saw any increase 
+in backpressure, given that no there was no backlog of records, and no 
+significant increase in the write to import latency.
+
+The results for our new cluster are specified bellow in the tables, where 
+our starting cluster configuration is the G3-S HA Plan and the new 
+configuration cluster is the G3 - BasePackage HA.
+
+| G3-S HA                | CPU Limit | Memory Limit in GB |
+|------------------------|-----------|--------------------|
+| operate                | 2         | 2                  |
+| operate.elasticsearch  | 6         | 6                  |
+| optimize               | 2         | 2                  |
+| tasklist               | 2         | 2                  |
+| zeebe.broker           | 2.88      | 12                 |
+| zeebe.gateway          | 0.9       | 0.8                |
+| zeebeAnalytics         | 0.4       | 0.45               |
+| connectorBridge        | 0.4       | 0.512              |
+| **TOTAL**              | **16.58** | **25.762**         |
+
+| G3 - BasePackage HA   | CPU Limit | Memory Limit in GB |
+|-----------------------|-----------|--------------------|
+| operate               | 1         | 1                  |
+| operate.elasticsearch | 3         | 4.5                |
+| optimize              | 1         | 1.6                |
+| tasklist              | 1         | 1                  |
+| zeebe.broker          | 1.5       | 4.5                |
+| zeebe.gateway         | 0.6       | 1                  |
+| zeebeAnalytics        | 0.2       | 0.3                |
+| connectorBridge       | 0.4       | 1                  |
```

Comment:
:wrench: Would remove in both tables, nobody outside of SaaS knows this, it's pretty internal